packages/bun-usockets/src/crypto/openssl.c

```diff
@@ -1693,7 +1695,7 @@ void *us_internal_ssl_socket_ext(struct us_internal_ssl_socket_t *s) {
 
 int us_internal_ssl_socket_is_shut_down(struct us_internal_ssl_socket_t *s) {
   return us_socket_is_shut_down(0, &s->s) ||
-         SSL_get_shutdown(s->ssl) & SSL_SENT_SHUTDOWN;
+         (s->ssl && (SSL_get_shutdown(s->ssl) & SSL_SENT_SHUTDOWN));
```

Comment:
when is the s->ssl pointer going to be null?

Reply:
never this should not be commited will revert